### PR TITLE
Correct edpm.yml artifacts collection

### PR DESCRIPTION
**Pull request by cjeanner**
An error might be avoided by adding the needed `when` condition, and
there was an issue in the way we detect multinode case and go down that
specific path.

As a pull request owner and reviewers, I checked that:
- [X] Appropriate testing is done and actually running

### CODE REVIEW

**Comment by cjeanner** [comment]
/unhold

At last we get a clean log for that log gathering:
```
2023-09-04 04:46:04.680311 | controller | TASK [artifacts : Generate logs on edpm vm {{ item.split('/')[0] }} output_dir={{ cifmw_artifacts_basedir }}/artifacts, script=ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no \
2023-09-04 04:46:04.680321 | controller |   -i {{ ssh_key_file }} {{ ssh_user }}@{{ host_ip }} <<EOF
2023-09-04 04:46:04.680330 | controller | set -x
2023-09-04 04:46:04.680338 | controller | sudo dnf install -y rsync
2023-09-04 04:46:04.680346 | controller | mkdir -p /tmp/{{ host_ip }}
2023-09-04 04:46:04.680354 | controller | sudo cp -a /var/log/ /tmp/{{ host_ip }}
2023-09-04 04:46:04.680363 | controller | sudo test -d /var/lib/openstack && sudo cp -a /var/lib/openstack /tmp/{{ host_ip }}
2023-09-04 04:46:04.680379 | controller | sudo test -d /var/lib/config-data && sudo cp -a /var/lib/config-data /tmp/{{ host_ip }}
2023-09-04 04:46:04.680388 | controller | sudo test -d /var/lib/cloud && sudo cp -a /var/lib/cloud /tmp/{{ host_ip }}
2023-09-04 04:46:04.680396 | controller | sudo test -d /etc/nftables && sudo cp -a /etc/nftables /tmp/{{ host_ip }}
2023-09-04 04:46:04.680404 | controller | sudo test -d /etc/os-net-config && sudo cp -a /etc/os-net-config /tmp/{{ host_ip }}
2023-09-04 04:46:04.680412 | controller | sudo find /tmp/{{ host_ip }} -type d -exec chmod ugoa+rx '{}' \;
2023-09-04 04:46:04.680420 | controller | sudo find /tmp/{{ host_ip }} -type f -exec chmod ugoa+r '{}' \;
2023-09-04 04:46:04.680428 | controller | command -v ovs-vsctl && sudo ovs-vsctl list Open_vSwitch > /tmp/{{ host_ip }}/ovs_vsctl_list_openvswitch.txt
2023-09-04 04:46:04.680437 | controller | sudo ip netns > /tmp/{{ host_ip }}/ip_netns.txt
2023-09-04 04:46:04.680445 | controller | sudo ip a > /tmp/{{ host_ip }}/network.txt
2023-09-04 04:46:04.680453 | controller | sudo ip ro ls >> /tmp/{{ host_ip }}/network.txt
2023-09-04 04:46:04.680461 | controller | sudo rpm -qa > /tmp/{{ host_ip }}/rpm_qa.txt
2023-09-04 04:46:04.680490 | controller | sudo podman images > /tmp/{{ host_ip }}/podman_images.txt
2023-09-04 04:46:04.680501 | controller | sudo ausearch -i | grep denied > /tmp/{{ host_ip }}/selinux-denials.log || true
2023-09-04 04:46:04.680510 | controller | sudo journalctl -p warning -t kernel -o short -g DROPPING --no-pager &> /tmp/{{ host_ip }}/firewall-drops.txt || true
2023-09-04 04:46:04.680518 | controller | EOF] ***
2023-09-04 04:46:04.680526 | controller | Monday 04 September 2023  04:46:04 -0400 (0:00:00.108)       0:00:24.749 ******
2023-09-04 04:46:10.150183 | controller | changed: [localhost] => (item=192.168.122.100)
2023-09-04 04:46:10.150252 | controller |
2023-09-04 04:46:10.150270 | controller | TASK [artifacts : Copy logs to host machine from {{ item.split('/')[0] }} output_dir={{ cifmw_artifacts_basedir }}/artifacts, script=rsync -a -e "ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i {{ ssh_key_file }}" {{ ssh_user }}@{{ host_ip }}:/tmp/{{ host_ip }} {{ cifmw_artifacts_basedir }}/logs/] ***
2023-09-04 04:46:10.150281 | controller | Monday 04 September 2023  04:46:10 -0400 (0:00:05.469)       0:00:30.218 ******
2023-09-04 04:46:10.768952 | controller | changed: [localhost] => (item=192.168.122.100)
```

No more error. Let's go!

File: ci_framework/roles/artifacts/tasks/edpm.yml [hunk 1]
```diff
@@ -10,78 +17,91 @@
...
+            dir: /etc/ci/env/
 
         - name: Extract Compute from zuul mapping if any
           when:
             - crc_ci_bootstrap_networks_out is defined
-            - cifmw_edpm_deploy_extra_vars
+            - cifmw_edpm_deploy_extra_vars is defined
           ansible.builtin.set_fact:
             ssh_key_file: "{{ cifmw_edpm_deploy_extra_vars.SSH_KEY_FILE }}"
+            ssh_user: zuul
```
Comment (pablintino) [inline ci_framework/roles/artifacts/tasks/edpm.yml:50]: This one is safe, but we are using this value before hardcoding it to zuul: https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/hooks/playbooks/fetch_compute_facts.yml#L168

Comment (cjeanner) [reply ci_framework/roles/artifacts/tasks/edpm.yml:50]: wondering about multiple compute support though... Maybe `{{ ansible_user }}` would be enough actually?

**Pull request merged**